### PR TITLE
Create a method to quality check (keep current behaviour)

### PR DESCRIPTION
**Pull request by fernandog**
@ratoaq2

### CODE REVIEW

File: medusa/common.py [hunk 1]
```diff
@@ -584,6 +584,23 @@ def statusFromName(name, anime=False):
...
+
+        if cur_status != WANTED:
+            if preferred_qualities:
+                if cur_quality in preferred_qualities:
+                    return False
+            elif cur_quality in allowed_qualities:
+                return False
+        return True
+
+    @staticmethod
```
Comment (fernandog) [inline medusa/common.py]: @pymedusa/developers wondering in UNKNOWN should be in this list? afaik if quality is unknown, ANY new quality is wanted right ?

File: medusa/post_processor.py [hunk 2]
```diff
@@ -1009,15 +1009,14 @@ def process(self):
                 self._log(u'File exists and the new file has the same size, aborting post-processing')
                 return True
 
-            if all([new_ep_quality <= old_ep_quality,
+            if all([new_ep_quality <= old_ep_quality, # Why we check if this ?!
```
Comment (fernandog) [inline medusa/post_processor.py]: @pymedusa/developers any one knows why we check if `new_ep_quality <= old_ep_quality` ?

**Comment by fernandog** [comment]
Last commit is not mandatory. It needs JS to proper work so as soon users changes, it updates.

any

File: medusa/common.py [hunk 3]
```diff
@@ -584,6 +584,23 @@ def statusFromName(name, anime=False):
     }
 
     @staticmethod
+    def is_wanted_quality(status, show_obj):
+        """Return true if that quality is wanted based on show quality and episode status."""
+        cur_status, cur_quality = Quality.splitCompositeStatus(int(status) or -1)
+        allowed_qualities, preferred_qualities = show_obj.current_qualities
+
+        if cur_status not in {WANTED, DOWNLOADED, SNATCHED, SNATCHED_PROPER}:
```
Comment (ratoaq2) [inline medusa/common.py]: You can change it to tuple. That will not change the logic, it will have the exact same behaviour

File: medusa/tv.py [hunk 4]
```diff
@@ -1839,33 +1840,10 @@ def get_overview(self, ep_status):
 
         if ep_status == WANTED:
             return Overview.WANTED
-        elif ep_status in (UNAIRED, UNKNOWN):
```
Comment (ratoaq2) [inline medusa/tv.py]: What changed here?

Comment (fernandog) [inline medusa/tv.py]: @ratoaq2 backlogOverview never showed unaired|unknown. It only shows wanted and low quality

**Comment by fernandog** [comment]
@ratoaq2 now there is only two parts of code that check preferred_qualities:


One is a simple called `is_final_result` and the other is this new method
image

**Comment by fernandog** [comment]
@pymedusa/developers can anyone review this please?

File: medusa/common.py [hunk 5]
```diff
@@ -584,6 +584,23 @@ def statusFromName(name, anime=False):
     }
 
     @staticmethod
+    def should_search(status, show_obj):
+        """Return true if that episodes should be search for a better quality."""
+        cur_status, cur_quality = Quality.splitCompositeStatus(int(status) or -1)
```
Comment (ratoaq2) [inline medusa/common.py]: Magic number here.
`-1` refers to UNKNOWN, right?

**Comment by fernandog** [comment]
@ratoaq2 anything else?

**Comment by fernandog** [comment]
i replaced, didn't do find usages. im using this branch since day1

**Pull request merged**